certif/app/components/enrolled-candidates.js

```diff
@@ -99,27 +99,32 @@ export default class EnrolledCandidates extends Component {
     this.notifications.clearAll();
     const certificationCandidate = this._createCertificationCandidateRecord(certificationCandidateData);
 
-    if (this._hasDuplicate(certificationCandidate)) {
-      this._handleDuplicateError(certificationCandidate);
-      return;
-    }
-
     try {
       await certificationCandidate.save({
         adapterOptions: { registerToSession: true, sessionId: this.args.sessionId },
       });
       this.args.reloadCertificationCandidate();
       this.notifications.success(this.intl.t(`${TRANSLATE_PREFIX}.add-modal.notifications.success-add`));
       return true;
-    } catch (err) {
-      if (this._hasConflict(err)) {
-        this._handleDuplicateError(certificationCandidate);
-      } else if (this._isEntityUnprocessable(err)) {
-        this._handleEntityValidationError(certificationCandidate, err);
-      } else {
-        this._handleUnknownSavingError(certificationCandidate);
+    } catch (errorResponse) {
+      const status = get(errorResponse, 'errors[0].status');
+      let errorText;
+
+      switch (status) {
+        case '409':
+          errorText = this._handleDuplicateError();
+          break;
+        case '422':
+          errorText = this._handleEntityValidationError(errorResponse);
+          break;
+        case '400':
+          errorText = this._handleMissingQueryParamError(errorResponse);
+          break;
+        default:
+          errorText = this._handleUnknownSavingError();
+          break;
```

Comment:
```suggestion
      function getErrorText(errorResponse) {
        const status = get(errorResponse, 'errors[0].status');
        
        switch (status) {
          case '409':
            return this._handleDuplicateError();
          case '422':
            return this._handleEntityValidationError(errorResponse);
          case '400':
            return this._handleMissingQueryParamError(errorResponse);
          default:
          return this._handleUnknownSavingError();
        } 
      }
      this._handleSavingError(getErrorText(errorResponse), certificationCandidate);
      return false
```

à voir si ça a du sens de garder les méthodes privées ou si l'on fait directement des appels à `intl`